### PR TITLE
Dash 1.0 breaking changes (DO NOT MERGE until Dash 1.0 is released)

### PR DESCRIPTION
**Pull request by alexcjohnson**
Breaking changes we're making for the Dash 1.0 release https://github.com/plotly/dash/issues/469
- `serve_locally=True` - https://github.com/plotly/dash/pull/722
- `show_undo_redo` - https://github.com/plotly/dash/pull/724
- Added migration guide

Post-merge checklist:

The master branch is auto-deployed to `dash.plot.ly`.
Once you have merged your PR, wait 5-10 minutes and check dash.plot.ly 
to verify that your changes have been made.

- [x] I understand

### CODE REVIEW

**Comment by alexcjohnson** [comment]
Added the hidden undo/redo buttons - we can make this PR the home for all dash 1.0 breaking changes.

File: tutorial/migration.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    ## `dash_table`
+
+    Lots of properties are renamed and/or restructured for better
+    self-consistency and extensibility.
+
+    | Old Prop         | NewProp             | Notes |
+    |------------------|---------------------|-------|
+    |`column.clearable`|`dropdown*.clearable`|`clearable` is a property of dropdowns, not really of columns. So it has moved into `dropdown`, `dropdown_conditional`, and `dropdown_data`.|
+    |`column.deletable`|                     |For multi-line headers, accepts an array of booleans rather than a number. For example, if there are multiple headers and you want the second header row to be deletable, this would be `[False, True]`.|
+    |`column.hidden`   |**REMOVED**          |Will be added back later in a different form. The main use case for this was covered by row IDs anyway.|
```
Comment (Marc-Andre-Rivet) [inline tutorial/migration.py]: `Notes` looks like copy-paste. Cross-link to https://github.com/plotly/dash-table/issues/314 instead.

File: tutorial/migration.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    With it we are affirming the maturity and stability of the Dash and
+    Dash component API. After making this migration, which we expect will
+    impact a sizable fraction of Dash users in one way or another, Dash will
+    not introduce any breaking changes unless and until we bump the major
+    version again. That commitment includes the public API of the whole Dash
+    "core": `dash`, `dash_core_components`, `dash_html_components`, and
+    `dash_table`.
+
+    Here are the changes Dash developers need to know about to transition apps
+    from v0.44 (the last release in the v0.x series) to v1.0.
```
Comment (Marc-Andre-Rivet) [inline tutorial/migration.py]:
```suggestion
    from v0.43 (the last release in the v0.x series) to v1.0.
```

File: tutorial/faqs.py [hunk 3]
```diff
@@ -135,6 +135,19 @@ class names of your components. Both `dash-html-components` and
 
     ------------------------
 
+    **Q:** *Where did those cool undo and redo buttons go?*
+
+    **A:** OK, mostly we got the opposite question:
```
Comment (chriddyp) [inline tutorial/faqs.py]: 😺

File: tutorial/migration.py (hunk 1 above)
Comment (alexcjohnson) [reply tutorial/migration.py]: good call -> https://github.com/plotly/dash-docs/pull/500/commits/8f3285ccde325dd127c7267de4e80079c8851c91

**Pull request merged**